pkg/insights/insightsreport/insightsreport.go

```diff
@@ -0,0 +1,164 @@
+package insightsreport
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/RedHatInsights/insights-results-smart-proxy/types"
```

Comment:
I'm a bit ambivalent here: OTOH I see the motivation of having one place where the structure is defined, on the other, this brings quite a big package of vendors with it. What do you think @tisnik? 

An altnerative would be defining the type here, with only the fields we're intersted in here. The benefit would seeing immediately what data are used. The downside is type control not telling us that in case something changers on the definition side. OTOH once this gets to real clusters we would need more or less freeze the changes there anyway, as getting the changes though to the clusters would take some time.

Reply:
@iNecas my personal feeling is that we might copy the structure into IO operator code. And Rob Pike seems not to disagree https://www.youtube.com/watch?v=PAAkCSZUG1c&t=9m28s

There are cons, of course, but the structure itself is pretty stable right now.

Reply:
As a newcomer in the Golang world, I would like to follow the best practices possible in the Golang ecosystem, so if you think that copying the struct is better than referencing it, I will do it on that way :-D